tests/integration/azure/test_locations.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+from databricks.labs.blueprint.installation import MockInstallation
+from databricks.sdk.errors.platform import NotFound
+
+from databricks.labs.ucx.azure.access import AzureResourcePermissions
+from databricks.labs.ucx.azure.locations import ExternalLocationsMigration
+from databricks.labs.ucx.azure.resources import AzureResources
+from databricks.labs.ucx.hive_metastore import ExternalLocations
+from databricks.labs.ucx.hive_metastore.locations import ExternalLocation
+
+
+def save_delete_location(ws, name):
+    try:
+        ws.external_locations.delete(name, force=True)
+    except NotFound:
+        # If test failed with exception threw before external location is created,
+        # don't fail the test with external location cannot be deleted error,
+        # instead let the original exception be reported.
+        pass
+
+
+@pytest.mark.skip
+def test_run(caplog, ws, sql_backend, inventory_schema):
+    locations = [
+        ExternalLocation("abfss://uctest@ziyuanqintest.dfs.core.windows.net/one", 1),
```

Comment:
Okay, as a follow-up after this PR is merged, make a PR to labs-infra which will setup these with Terraform. Or maybe within this PR. I'll do you a quick intro

Reply:
The storage account, managed identities etc. I'm using here belong to the field-eng subscription and they are subject to be removed with `removeAfter` tag.
If we use Terraform to populate them in labs-infra, the problem is I don't have read permission for Azure resources in labs-infra, those azure graph and resource manager api call to get application_id will fail.